umh-core/pkg/config/manager.go

```diff
@@ -686,3 +708,64 @@ func (m *FileConfigManagerWithBackoff) AtomicEditDataflowcomponent(ctx context.C
 
 	return m.configManager.AtomicEditDataflowcomponent(ctx, componentUUID, dfc)
 }
+
+// GetConfigAsString returns the current config file contents as a string
+func (m *FileConfigManager) GetConfigAsString(ctx context.Context) (string, error) {
+	// in the GetConfig method, we already read the file and cached the raw config to m.cacheRawConfig
+	_, err := m.GetConfig(ctx, 0)
```

Comment:
i don't understand this, why are we doing a GetConfig and then reading from the cache? that doesn't feel right, even if this works. I recommend splitting it then up into functions that mutate state, and then pure functions (as one example). Otherwise it is unexpected.